deadline-react/js/components/challenges/ChallengesPage.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Challenge from './Challenge'
+import { Link } from 'react-router-dom'
+import { getSubCategoryChallenges, getCategorySubcategories } from '../../requests'
+import { convertFromUrlToFriendlyText } from '../../helpers'
+import SelectionSearch from '../../semantic_ui_components/SelectionSearch.js'
+
+class ChallengesPage extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      subcategories: [],
+      challenges: []
+    }
+
+    this.loadSubcategoryChallenges = this.loadSubcategoryChallenges.bind(this)
+    this.loadSubcategories = this.loadSubcategories.bind(this)
+    this.loadSubcategoryChallenges()
+    this.loadSubcategories()
+  }
+
+  loadSubcategoryChallenges () {
+    const subcategory = convertFromUrlToFriendlyText(this.props.match.params.subcategory)
+    console.log('Querying with ' + subcategory)
+    getSubCategoryChallenges(subcategory).then(subcat => {
+      let challenges = subcat.challenges
```

Comment:
idk if you need to have challenges as a separate variable here, as it does not make the code clearer or anything, just takes a line
Actually I understand you're doing it to write `{challenges}` as opposed to `{challenges: subcat.challenges}`. IMO explicit is better than implicit but this isn't that important